### PR TITLE
Implement ONC CareTeam, Fixed PractitionerRole, Fix critical bug in lookup_code_information

### PR DESCRIPTION
**Pull request by adunsulag**
Implemented the care team ONC requirements.  Got the queries working on
the practitioner role.

Also fixed a bug with lookup_code_information that got introduced due to a bad rebase in the prior ONC commitment.

We need to implement a status column in the care team on OpenEMR gui before we can fully check this off.
See #4405 
image

### CODE REVIEW

File: _rest_routes.inc.php [hunk 1]
```diff
@@ -886,14 +886,21 @@
...
         RestConfig::apiLog($return);
         return $return;
     },
     "GET /fhir/Organization/:id" => function ($id, HttpRestRequest $request) {
-        RestConfig::authorization_check("admin", "users");
-        $return = (new FhirOrganizationRestController())->getOne($id);
+        $patientUUID = null;
+        if (!$request->isPatientRequest()) {
+            RestConfig::authorization_check("admin", "users");
+            $patientUUID = $request->getPatientUUIDString();
```
Comment (bradymiller) [inline _rest_routes.inc.php:900]: this is throwing me off a bit. Isn't this block for when not using a patient request?

Comment (bradymiller) [reply _rest_routes.inc.php:900]: btw,
Highly recommend for folks to change github to dark theme while playing this youtube song at same time :) :
https://www.youtube.com/watch?v=X2LTL8KgKv8

image

Comment (stephenwaite) [reply _rest_routes.inc.php:900]: looks like inferno needs to give in to the dark side

Comment (bradymiller) [reply _rest_routes.inc.php:900]: and we can never forget this openemr classic :)
https://www.youtube.com/watch?v=xSmYAdiXb5M

Comment (stephenwaite) [reply _rest_routes.inc.php:900]: never :) mesmerized by the speed bag :boxing_glove:

File: src/Services/ImmunizationService.php [hunk 2]
```diff
@@ -119,141 +108,103 @@ public function getAll($search = array(), $isAndCondition = true, $puuidBind = n
...
 
-        $isValid = $this->immunizationValidator->validateId("uuid", "immunizations", $uuid, true);
-        if ($isValid !== true) {
-            $validationMessages = [
-                'uuid' => ["invalid or nonexisting value" => " value " . $uuid]
-            ];
-            $processingResult->setValidationMessages($validationMessages);
-            return $processingResult;
+        if (isset($search['patient.uuid'])) {
+        }
```
Comment (bradymiller) [inline src/Services/ImmunizationService.php:150]: a residual if block here

**Comment by bradymiller** [comment]
hi @adunsulag , Code review complete. Only had 1 question and noted a very minimal issue noted above. This PR will be ready after addressed :)

**Comment by adunsulag** [comment]
@bradymiller This should now be fixed with my latest changes.

**Comment by bradymiller** [comment]
@adunsulag , looks good and bringing in. thanks for the fhir!! -brady

2DV (1)

**Pull request merged**